ingest/expression_files/expression_files.py

```diff
@@ -93,17 +93,25 @@ def create_gene_model(
         )
 
     @staticmethod
-    def check_unique_cells(cell_names: List, study_id, client):
-        """Checks cell names against database to confirm matrix contains unique
-            cell names.
+    def is_raw_count(study_file_id, client):
+        COLLECTION_NAME = "study_files"
+        FIELD_NAMES = {"expression_file_info.is_raw_counts": 1, "_id": 0}
+        QUERY = {"_id": study_file_id}
+        query_results = list(client[COLLECTION_NAME].find(QUERY, FIELD_NAMES)).pop()
+        return query_results["expression_file_info"]["is_raw_counts"]
+
+    @staticmethod
+    def check_unique_cells(cell_names: List, study_id, study_file_id, client):
```

Comment:
I think this method as a whole would be easier to read and test if you removed all mongo query logic from it, so this method just reads as a list of logical steps, and the mongo interfacing is delegated out.  E.g. 

```
check_unique_cells(cell_names, study_id, study_file_id, client):
    study_file = get_study_file_by_id(study_file_id)
    is_raw_counts = is_raw_counts_file(study_file)
    cells_of_same_type = get_study_cells_by_type(study_id, is_raw_counts)
    <<compare cells_of_same_type and cell_names for dupes>>

```

Reply:
I'm not convinced that it would be easier to test. The same methods still need to be mocked. The query also needs to be validated for every test case until we have a mongo emulator. 

However, I did abstract out the mongo query logic and the code does look cleaner! Thanks for making the suggestions. 

99313db

Reply:
[SCP-2000](https://broadworkbench.atlassian.net/browse/SCP-2000) tracks integrating a MongoDB emulator.